src/data-model-providers/codedriven/EndpointProviderInterface.h

```diff
@@ -0,0 +1,59 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/data-model-provider/MetadataTypes.h>
+#include <app/server-cluster/ServerClusterInterface.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/support/ReadOnlyBuffer.h>
+
+namespace chip {
+namespace app {
+
+using SemanticTag = chip::app::Clusters::Descriptor::Structs::SemanticTagStruct::Type;
+
+/**
+ * @brief Defines the interface for an object that can provide information about a Matter endpoint.
+ *
+ * This interface is used by the CodeDrivenDataModelProvider to discover the structure and capabilities
```

Comment:
Maybe we should not link the CodeDriveDatamodelProvider but rathe document what this class is intended to provide as a stand-alone. Anyone can use it for any purpose.